test/1_stdlib/SetAlgebra.swift

```diff
@@ -0,0 +1,188 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+public struct DefaultedSetAlgebra<Element> : SetAlgebra {
+  public typealias Element = String
+  var store = Set<String>()
+  public init(){}
+  public init(_ elements: [Element]) {
+    for el in elements {
+      store.insert(el)
+    }
+  }
+
+  public func contains(_ member: String) -> Bool {
+    return store.contains(member)
+  }
+
+  public func union(_ other: DefaultedSetAlgebra) -> DefaultedSetAlgebra {
+    var defaultedSetAlgebra = DefaultedSetAlgebra()
+    defaultedSetAlgebra.store = store.union(other.store)
+    return defaultedSetAlgebra
+  }
+
+  public func intersection(_ other: DefaultedSetAlgebra) -> DefaultedSetAlgebra {
+    var defaultedSetAlgebra = DefaultedSetAlgebra()
+    defaultedSetAlgebra.store = store.intersection(other.store)
+    return defaultedSetAlgebra
+  }
+
+  public func symmetricDifference(_ other: DefaultedSetAlgebra) -> DefaultedSetAlgebra {
+    var defaultedSetAlgebra = DefaultedSetAlgebra()
+    defaultedSetAlgebra.store = store.symmetricDifference(other.store)
+    return defaultedSetAlgebra
+  }
+
+  public mutating func insert(_ newMember: String) -> (inserted: Bool, memberAfterInsert: String) {
+    return store.insert(newMember)
+  }
+
+  public mutating func remove(_ member: String) -> String? {
+    return store.remove(member)
+  }
+
+  public mutating func update(with newMember: String) -> String? {
+    return store.update(with: newMember)
+  }
+
+  public mutating func formUnion(_ other: DefaultedSetAlgebra) {
+    store.formUnion(other.store)
+  }
+
+  public mutating func formSymmetricDifference(_ other: DefaultedSetAlgebra) {
+    store.formSymmetricDifference(other.store)
+  }
+
+  public mutating func formIntersection(_ other: DefaultedSetAlgebra) {
+    store.formIntersection(other.store)
+  }
+}
+
+public func ==<Element>(lhs: DefaultedSetAlgebra<Element>, rhs: DefaultedSetAlgebra<Element>) -> Bool {
+  return lhs.store == rhs.store
+}
+
+let SetAlgebraTests = TestSuite("SetAlgebra")
+
+SetAlgebraTests.test("contains") {
+  let primes = DefaultedSetAlgebra<String>(["2", "3", "5", "7"])
+  expectTrue(primes.contains("5"))
+  expectFalse(primes.contains("6"))
+}
+
+SetAlgebraTests.test("union") {
+  let attendees = DefaultedSetAlgebra<String>(["Alicia", "Bethany", "Diana"])
+  let visitors = DefaultedSetAlgebra<String>(["Marcia", "Nathaniel"])
+  expectEqual(["Diana", "Nathaniel", "Bethany", "Alicia", "Marcia"],
+    attendees.union(visitors))
+
+  let initialIndices = DefaultedSetAlgebra<String>(["0", "1", "2", "3", "4"])
+  expectEqual(DefaultedSetAlgebra<String>(["2", "4", "6", "7", "0", "1", "3"]),
+    initialIndices.union(DefaultedSetAlgebra<String>(["2", "3", "6", "7"])))
```

Comment:
(This test and others)

Would you mind expanding these tests to be data-driven, like `findTests` and its uses that I referenced above?  This way it is really easy to add more test cases for the same function, without the repetitive construct-act-check structure of the test, and we can reuse the same dataset for all types that conform to `SetAlgebra`.

For example, the `findTests` dataset is used 7 times in the Swift codebase.

If you feel like you don't want to implement this change, that is perfectly fine, just let me know, and we will get it merged.